src/engine/helper.ts

```diff
@@ -0,0 +1,87 @@
+import Square from "./square";
+import GameSettings from "./gameSettings";
+import Board from "./board";
+
+export function moveInBounds(newMove: Square) {
+    if (
+        (newMove.col > GameSettings.BOARD_SIZE-1) ||
+        (newMove.col < 0)||
+        (newMove.row > GameSettings.BOARD_SIZE-1) ||
+        (newMove.row < 0)
+    ){
+        return false;
+    } 
+    return true;
+}
+
+export function getDiagonalMoves(board:Board, currentPosition: Square){
+    let moveList = []
+
+    const directions = [
+        [1, 1],
+        [-1, -1],
+        [-1, 1],
+        [1, -1]
+    ];
+
+    moveList = getMovesFromDirections(board, currentPosition, directions);

```

Comment:
```suggestion
    const moveList = getMovesFromDirections(board, currentPosition, directions);
```

no need to define and immediately reassign